shim/linux/src/entry.c

```diff
@@ -225,10 +226,53 @@ dispatch_system_kvm_get_msr_feature_index_list(
 }
 
 static long
-dispatch_system_kvm_get_msr_index_list(struct kvm_msr_list *const ioctl_args)
+dispatch_system_kvm_get_msr_index_list(
+    struct kvm_msr_list __user *const user_args)
 {
-    (void)ioctl_args;
-    return -EINVAL;
+    struct kvm_msr_list mut_args;
+    uint32_t __user *pmut_mut_user_indices;
+    int64_t mut_ret;
+
+    mut_ret = -EINVAL;
+    if (platform_copy_from_user(&mut_args, user_args, sizeof(mut_args))) {
+        bferror("platform_copy_from_user failed");
+        goto out;
+    }
+
+    pmut_mut_user_indices = mut_args.indices;
+    mut_args.indices = vzalloc(mut_args.nmsrs * sizeof(*mut_args.indices));
```

Comment:
Looking at this again, it seems that the only logic that could move from the dispatch function would be `vzalloc`, because in order to call `vfree` in the handle function, we would need a `copy_to_user` call in the handle function also.

It seems to me that moving `vzalloc` and/or `copy_to_user` would only (1) complicate unit testing, and (2) make this code less clear.